addons/stock/models/stock_picking.py

```diff
@@ -432,6 +437,119 @@ def _compute_show_picking_type(self):
         for record in self:
             record.show_picking_type = record.code in ['incoming', 'outgoing', 'internal']
 
+    def _get_date_aggregated_records(
+        self, picking_types, model_name='stock.picking', required_state='assigned',
+        date_field_name='scheduled_date', data_series_name=_('Ready')
+    ):
+        """
+        Returns a list, each element containing 3 values:
+        * picking type ID
+        * list of date fields values of all pickings with that picking type
+        * data series name, used to display it in the graph
+        """
+        records = self.env[model_name]._read_group(
+            [
+                ('company_id', 'in', self.env.companies.ids),
+                ('picking_type_id', 'in', picking_types.ids),
+                ('state', '=', required_state)
+            ],
+            ['picking_type_id'],
+            [date_field_name + ':array_agg'],
+        )
+        # Append data series name to each record
+        return [(r[0], r[1], data_series_name) for r in records]
+
+    def _get_picking_type_dashboard_summaries(self, picking_types):
```

Comment:
Each picking type will have its own DB query. Those queries have to be applied only to the specific picking type, and all of the remaining picking types have to be processed by a different module.

Initially I used `self` here, but the result was that e.g. `mrp` module would process MOs, then `repair` would override the computation done by `mrp`, because from its perspective MOs are not a "repair" type, so they would be forwarded to the default `stock` code. This was a really annoying bug to find :P  

The solution I came up with was to split picking types into ones that should be processed in the current module and others, and only pass others for further processing. I am of course open for discussion, if you can suggest a better solution :)